coding/zlib.cpp

```diff
@@ -25,7 +25,7 @@ ZLib::Processor::Processor(void const * data, size_t size) noexcept : m_init(fal
   // zconf.h. So, for portability, const_cast<...> is used here, but
   // in any case, zlib does not modify |data|.
   m_stream.next_in = static_cast<unsigned char *>(const_cast<void *>(data));
-  m_stream.avail_in = size;
+  m_stream.avail_in = static_cast<unsigned int>(size);
```

Comment:
Probably we need to chagne Processor ctor signature, or, modify the code to work even on sizes that are greater than UINT_MAX. But not now.